Tests/HarmonizeTests/Tests/LintRulesExamplesTests.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  LintRulesExamplesTests.swift
+//  Harmonize
+//
+//  Copyright (c) Perry Street Software 2024. All Rights Reserved.
+//
+
+import Foundation
+import Harmonize
+import XCTest
+
+final class LintRulesExamplesTests: XCTestCase {
+    override var testRunClass: AnyClass? {
+        return AssertionsFailuresTests.ExpectedFailureTestCaseRun.self
+    }
+    
+    func testNoSideEffects() throws {
+        Harmonize.on { Self.ViewModel }
+            .classes()
+            .initializers()
+            .withFunctionCalls { functionCalls in
+                let closures = functionCalls.compactMap(\.closure)
+                let hasAssignmentToSelf = closures.filter {
+                    $0.body?.assignments.contains { $0.target.contains("self") } == true
```

Comment:
If we're checking `hasAssignmentToSelf`, shouldn't we check if the left operand (target) has the `self` keyword?

Reply:
Isn't it what we do with `$0.target.contains("self")`?

Reply:
Ah lol sorry missed that!